fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/AccountNumberGenerator.java

```diff
@@ -134,11 +147,50 @@ private String generateAccountNumber(Map<String, String> propertyMap, AccountNum
                 prefix = prefix.substring(0, Math.min(prefix.length(), 10));
             }
 
+            if (accountNumberPrefixType.getValue() == 401) {
+
+                AccountNumberFormat accountNumberByPrefix = this.accountNumberFormatRepository
+                        .findOneByPrefixEnumAccountTypeEnum(accountNumberFormat.getAccountTypeEnum(), 401);
+                Integer digits = accountNumberByPrefix.getDigits();
+                Integer numberLength = accountMaxLength - digits;
+                accountNumber = StringUtils.leftPad(propertyMap.get(ID), numberLength, '0');
+
+            } else {
+                accountNumber = StringUtils.leftPad(accountNumber, Integer.valueOf(propertyMap.get(ID).length()), '0');
+            }
+
             accountNumber = StringUtils.overlay(accountNumber, prefix, 0, 0);
         }
         return accountNumber;
     }
 
+    private Map<String, String> prefixEight(Map<String, String> propertyMap, String accountNumber, Integer accountMaxLength,
+            AccountNumberFormat accountNumberFormat) {
+        AccountNumberFormat accountNumberByPrefix = this.accountNumberFormatRepository
+                .findOneByPrefixEnumAccountTypeEnum(accountNumberFormat.getAccountTypeEnum(), 401);
+        Integer digits = accountNumberByPrefix.getDigits();
```

Comment:
Do we need to pass in digits as a separate argument and store it separately in the database? Could we not infer the length of the prefix from the length of the prefix string? Ie if we want the prefix to be 3 characters of "123" we just store "123" as the prefix string.